landoui/treestatus.py

```diff
@@ -126,6 +127,10 @@ def update_treestatus(api: TreestatusAPI, update_trees_form: TreeStatusUpdateTre
     Treestatus page on success. Display an error message and return to the form if
     the status updating rules were broken or the API returned an error.
     """
+    if not is_user_authenticated():
+        flash("Authentication is required to update tree statuses.")
+        return redirect(request.referrer)
```

Comment:
As mentioned in the bug, I think this should return a `401` status code. You can add a `code` attribute to the redirect call. This applies to all other instances of this change in this PR.